contractsV5/FeeHandler.sol

```diff
@@ -0,0 +1,189 @@
+pragma solidity 0.5.11;
+
+import "./IKyberDAO.sol";
+import "./IFeeHandler.sol";
+import "./PermissionGroupsV5.sol";
+import "./KyberNetworkProxy.sol";
+import "./UtilsV5.sol";
+import "./IERC20.sol";
+
+contract FeeHandler is IFeeHandler, Utils {
+
+    IKyberDAO public kyberDAOContract;
+    // if I use the interface, I am not able to get the KyberNetwork internal contract from the proxy..
+    KyberNetworkProxy public kyberNetworkProxyContract;
+    address public knc;
+    address payable public burnAddress;
+    uint public burnBlockInterval;
+    uint public lastBurnBlock;
+    uint constant ETH_TO_BURN = 10**19;
+
+    uint public brrAndEpochData;
+    uint constant BITS_PER_PARAM = 64;
+
+    uint public totalRebates;
+    mapping(address => uint) public totalRebatesPerRebateWallet;
+    uint public totalRewards;
+    mapping(uint => uint) public totalRewardsPerEpoch;
+
+
+    constructor(
+        IKyberDAO _kyberDAOContract,
+        KyberNetworkProxy _kyberNetworkProxyContract,
+        address _knc,
+        address payable _burnAddress,
+        uint _burnBlockInterval
+    ) public
+    {
+        kyberDAOContract = _kyberDAOContract;
+        kyberNetworkProxyContract = _kyberNetworkProxyContract;
+        knc = _knc;
+        burnAddress = _burnAddress;
+        burnBlockInterval = _burnBlockInterval;
+        lastBurnBlock = block.number;
+    }
+
+    modifier onlyDAO {
+        require(
+            msg.sender == address(kyberDAOContract),
+            "Only the DAO can call this function."
+        );
+        _;
+    }
+
+    modifier onlyKyberNetwork {
+        require(
+            msg.sender == address(kyberNetworkProxyContract.kyberNetworkContract()),
+            "Only the internal KyberNetwork contract can call this function."
+        );
+        _;
+    }
+
+
+    function encodeData(uint _burn, uint _reward, uint _epoch, uint _expiryBlock) public pure returns (uint) {
+        return (((((_burn << BITS_PER_PARAM) + _reward) << BITS_PER_PARAM) + _epoch) << BITS_PER_PARAM) + _expiryBlock;
```

Comment:
nice functions. (encode, decode)
but remove them down in the file.
main logic better be on top.

these are like helper functions.

Reply:
noted and updated, will push when the other comments are fixed.